### DISCUSSION TITLE
怎样通过Java实现基于v2ray开发移动app

### DISCUSSION THREAD

**Discussion by MyNanyh** [v2fly/v2ray-core#2632] [Q&A]
是否有关于Java开发的交流群呢，或者有什么建议给到我吗谢谢

**Comment by ZXiangQAQ** [comment]
v2rayNG，现成的开源项目

**Reply by laaArmy** [reply]
这个内核不是Xray吗

**Reply by vivodi** [reply]
> 这个内核不是Xray吗

内核可以自行替换的，Xray就是一个[小丑](https://github.com/RPRX)闹分家的产物。详情可以看 https://github.com/XTLS/Go/issues/9 和 https://github.com/v2ray/v2ray-core/issues/2789
